File: Python/BinaryTreeRightSideView/BinaryTreeRightSideView.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
	# (42ms)
    def rightSideView(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        ans, queue = [], [root]
        if root:
        	while queue:
	        	size = len(queue)
	        	for i in xrange(size):
	        		now = queue[i]
	        		if now.left:
	        			queue.append(now.left)
	        		if now.right:
	        			queue.append(now.right)
	        		if i == size - 1:
	        			ans.append(now.val)
	        	queue = queue[size:]
        return ans

File: JavaScript/BinaryTreeRightSideView/BinaryTreeRightSideView.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * (156ms)
 * @param {TreeNode} root
 * @return {number[]}
 */
var rightSideView = function(root) {
    var ans = [];
    if(root != null) {
        var queue = [];
        queue.push(root);
        while (queue.length > 0) {
            var size = queue.length;
            for (let i = 0; i < size; i++) {
                var now = queue.shift();
                if (now.left != null)
                    queue.push(now.left);
                if (now.right != null)
                    queue.push(now.right);
                if (i == size - 1)
                    ans.push(now.val);
            }
        }
    }
    return ans;
};